### PR TITLE
Add initial implementation of the DocuSign Rooms connector

### LINKED ISSUE #5964: Implement DocuSign Rooms Connector
Description:
This is subtask of https://github.com/ballerina-platform/ballerina-library/issues/5909

The APIs of the connector are based on the following OAS specification
https://github.com/docusign/OpenAPI-Specifications/blob/master/rooms.rest.swagger-v2.json

### PR DESCRIPTION
**Pull request by Nuvindu**
## Purpose
Fixes: https://github.com/ballerina-platform/ballerina-library/issues/5964

### CODE REVIEW

File: ballerina/utils.bal [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    # Specifies whether arrays and objects should generate as separate fields
+    boolean explode = true;
+    # Specifies the custom content type
+    string contentType?;
+    # Specifies the custom headers
+    map<any> headers?;
+};
+
+enum EncodingStyle {
+    DEEPOBJECT, FORM, SPACEDELIMITED, PIPEDELIMITED
```
Comment (ThisaruGuruge) [inline ballerina/utils.bal]:
```suggestion
    DEEPOBJECT, 
    FORM, 
    SPACEDELIMITED,
     PIPEDELIMITED
```

File: ballerina/utils.bal [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+# + arrayName - Name of the field with arrays
+# + anyArray - Array to be serialized
+# + style - Defines how multiple values are delimited
+# + explode - Specifies whether arrays and objects should generate separate parameters
+# + return - Serialized array as a string
+isolated function getSerializedArray(string arrayName, anydata[] anyArray, string style = "form", boolean explode = true) returns string {
+    string key = arrayName;
+    string[] arrayValues = [];
+    if (anyArray.length() > 0) {
+        if (style == FORM && !explode) {
```
Comment (ThisaruGuruge) [inline ballerina/utils.bal]: Check other places.

```suggestion
        if style == FORM && !explode {
```

File: ballerina/utils.bal [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+isolated function getSerializedRecordArray(string parent, record {}[] value, string style = FORM, boolean explode = true) returns string {
+    string[] serializedArray = [];
+    if style == DEEPOBJECT {
+        int arayIndex = 0;
+        foreach var recordItem in value {
+            serializedArray.push(getDeepObjectStyleRequest(parent + "[" + arayIndex.toString() + "]", recordItem), "&");
+            arayIndex = arayIndex + 1;
+        }
+    } else {
+        if (!explode) {
```
Comment (ThisaruGuruge) [inline ballerina/utils.bal]:
```suggestion
        if !explode {
```

**Review by NipunaRanasinghe — COMMENTED** [review]
@Nuvindu is this PR is valid? If we have decided to deprioratize this connector, better to close the PR by adding a comment with the reason to do so

**Comment by Nuvindu** [comment]
It was decided to keep this connector on hold as it requires a authorization from DocuSign to enable DocuSign Rooms feature which we don't have the access yet. Therefore it is hard to test the Rooms APIs. And since there is no current requirement for this feature, we have put this on hold

**Pull request closed without merging**